scripts/provisioning/setupcmd.py

```diff
@@ -145,3 +148,72 @@ def phase_prereqs_validate(self, phase_name: str):
                                 files=_prereqs_confstore.get_config(f'{phase_name}>files'))
     except Exception as e:
       raise S3PROVError(f'ERROR: {phase_name} prereqs validations failed, exception: {e} \n')
+
+  def is_service_active(self, s3service):
+    """Return True if service is running."""
+    self.service = s3service
+
+    cmd = ['/bin/systemctl', 'status',  f'{self.service}']
+    handler = SimpleProcess(cmd)
+    res_op, res_err, res_rc = handler.run()
+    if res_rc != 0:
+      raise Exception(f"{cmd} failed with err: {res_err}, out: {res_op}, ret: {res_rc}")
+
+    res_op_list = res_op.split('\n')
+    for line in res_op_list:
+        if 'Active:' in line:
+           if '(running)' in line:
+               return True
+    return False
+
+  def shutdown_active_service(self):
+    """Stop service and return returncode."""
+    cmd = ['/bin/systemctl', 'stop',  f'{self.service}']
+    handler = SimpleProcess(cmd)
+    sys.stdout.write(f"shutting down {self.service}\n")
+    res_op, res_err, res_rc = handler.run()
+    if res_rc != 0:
+      raise Exception(f"{cmd} failed with err: {res_err}, out: {res_op}, ret: {res_rc}")
+    else:
+      return res_rc
+
+  def shutdown_service(self, s3service):
+    """Stop service and return returncode."""
```

Comment:
Pass a list - it can have one or more elements

Reply:
done